### PR TITLE
Store Orders Page: Specific Order UI

### PR DESCRIPTION
**Pull request by alexzhang1618**
# Description

What changes did you make? List all distinct problems that this PR addresses. Explain any relevant
motivation or context.
- Implemented examining the info of a specific order

Things I still need to do:
- [x] Discount price displays
- [ ] Flows for doing stuff (cancelling, rescheduling, etc etc)
- [ ] animation for opening a card? this might be difficult
- [x] The specific event card header orientation for mobile (the status symbol moves to the top)

## Changes

- [Fill in here]

# Type of Change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] Logistics Change (A change to a README, description, or dev workflow setup like
      linting/formatting)
- [ ] Continuous Integration Change (Related to deployment steps or continuous integration
      workflows)
- [ ] Other: (Fill In) 

# Testing

I have tested that my changes fully resolve the linked issue ...

- [x] locally on Desktop.
- [ ] locally on mobile - use https://ngrok.io to get a copy on a mobile device
- [ ] on the live deployment preview on Desktop.
- [ ] on the live deployment preview on Mobile.
- [ ] I have run and passed all new and existing Cypress tests. Add screenshots below.

# Checklist

- [x] I have performed a self-review of my own code.
- [x] I have followed the style guidelines of this project.
- [x] I have documented my code's `src/lib` functions and commented hard to understand areas
      anywhere else.
- [x] My changes produce no new warnings.

# Screenshots
Screenshot 2023-12-30 at 12 05 36 AM

Screenshot 2023-12-30 at 12 06 00 AM

### CODE REVIEW

File: src/components/store/OrderCard/index.tsx
Comment (SheepTester) [inline src/components/store/OrderCard/index.tsx]: hmm, why force at most one order card to be open at a time? [exclusive accordions hurt accessibility and usability](https://github.com/openui/open-ui/issues/925), and it'd also perhaps be easier just for each `OrderCard` to keep track of their own state

File: src/components/store/OrderSummary/index.tsx
Comment (SheepTester) [inline src/components/store/OrderSummary/index.tsx]: you can also do

```suggestion
  const updatedItems = Array.from(itemMap.values());
```

File: src/components/store/OrderSummary/index.tsx
Comment (SheepTester) [inline src/components/store/OrderSummary/index.tsx]: I don't think this line is necessary since you're mutating the original object

File: src/components/store/OrderSummary/index.tsx
Comment (SheepTester) [inline src/components/store/OrderSummary/index.tsx:79]: You could also let CSS do this for you, using `text-transform: capitalize;`, though it capitalizes every word instead of just the first letter

**Review by SheepTester — COMMENTED** [review]
I think the text contrast could be improved. Probably the background color should be dark blue for dark theme

image

maybe on mobile, the text should show below the image?

image

> * animation for opening a card? this might be difficult

yes it will be, i recommend putting this off for later

File: src/components/store/OrderSummary/index.tsx
Comment (alexzhang1618) [reply src/components/store/OrderSummary/index.tsx:79]: https://stackoverflow.com/questions/52165435/why-css-text-transform-capitalize-not-working

sadly, it doesn't decapitalize the rest of the word 😢 
It's still possible to do this thru CSS, but double cascading lowercase and capitalize seems less readable, so I think I'll keep it as is.

File: src/components/store/OrderCard/index.tsx
Comment (alexzhang1618) [reply src/components/store/OrderCard/index.tsx]: Done, I'm writing this with `useEffect` and `useState` so we can stagger the number of getOrder API calls to be on demand instead of all on page load.

**Comment by alexzhang1618** [comment]
Since the last review: 

- Fixed some color contrast stuff for order status
- Implemented the Discount price views
- Made each card openable and moved API code into the card
- Added mobile layout for status
- Added action buttons to view pickup event, reschedule, and cancel order
  - I'm going to wait until Modal is merged in #91 before implementing these since I want to use it for these buttons

Screenshot 2023-12-30 at 1 37 38 PM
Screenshot 2023-12-30 at 1 38 20 PM
Screenshot 2023-12-30 at 1 38 29 PM
Screenshot 2023-12-30 at 1 38 56 PM
Screenshot 2023-12-30 at 1 41 19 PM

**Review by farisashai — CHANGES_REQUESTED** [review]
image

can you fix the blue color on the orders card and dropdown and also the padding for the store navbar is inconsistent with the page padding

File: src/components/store/Diamonds/index.tsx
Comment (farisashai) [inline src/components/store/Diamonds/index.tsx]: probably cleaner for the future to have a discount active classmate rather than doing all of this online styling

File: src/components/store/Diamonds/index.tsx
Comment (farisashai) [inline src/components/store/Diamonds/index.tsx]:
```suggestion
          &nbsp;
```

File: src/components/store/OrderCard/index.tsx
Comment (farisashai) [inline src/components/store/OrderCard/index.tsx:44]: This looks like the same Boolean you defined on 41

File: src/components/store/OrderCard/style.module.scss
Comment (farisashai) [inline src/components/store/OrderCard/style.module.scss]:
```suggestion
      border-radius: 0.5rem 0.5rem 0 0;
```

**Comment by alexzhang1618** [comment]
fixed by adding to reset.scss, I also removed the change in EventCard since this covers it as well. padding also fixed too

**Review by farisashai — APPROVED** [review]

File: src/components/store/OrderCard/index.tsx
Comment (alexzhang1618) [reply src/components/store/OrderCard/index.tsx:44]: It's not quite the same, this makes sure we don't refetch if we've already done so `orderData === null` versus `orderData !== null`

**Pull request merged**
